src/math/minmax.h

```diff
@@ -1,29 +1,9 @@
 #ifndef MINMAX_H_
 #define MINMAX_H_
 
-inline int64_t min_int64(int64_t a, int64_t b);
-inline int64_t max_int64(int64_t a, int64_t b);
-inline size_t min_size(size_t a, size_t b);
-inline size_t max_size(size_t a, size_t b);
-
-inline int64_t min_int64(int64_t a, int64_t b)
-{
-    return a < b ? a : b;
-}
-
-inline int64_t max_int64(int64_t a, int64_t b)
-{
-    return a > b ? a : b;
-}
-
-inline size_t min_size(size_t a, size_t b)
-{
-    return a < b ? a : b;
-}
-
-inline size_t max_size(size_t a, size_t b)
-{
-    return a > b ? a : b;
-}
+#define min_int64(a, b) (a < b ? a : b)
+#define max_int64(a, b) (a > b ? a : b)
+#define min_size(a, b) (a < b ? a : b)
+#define max_size(a, b) (a > b ? a : b)
```

Comment:
`a` and `b` are evaluated twice. Try

```c
min_int64(printf("hello\n"), printf("world\n"));
```

Reply:
Oh.

Reply:
@TheKnarf welcome to the C preprocessor buddy! \o/